docs/admin/templates/managing-templates/devcontainers/add-devcontainer.md

```diff
@@ -0,0 +1,143 @@
+workspace. Envbuilder clones the repo and builds a container from the
+`devcontainer.json` specified in the repo.
+
+You can create template files through the Coder dashboard, CLI, or you can
+choose a template from the
+[Coder registry](https://registry.coder.com/templates?tag=devcontainer):
+
+<div class="tabs">
+
+## Dashboard
+
+1. In the Coder dashboard, select **Templates** then **Create Template**.
+1. To use a
+   [starter template](https://github.com/coder/coder/tree/main/examples/templates),
+   select **Choose a starter template**.
+
+   - Select **From scratch** to create a new template and enter information and
+     file contents manually.
+
+## CLI
+
+1. Use the `template init` command to initialize your choice of image:
+
+   ```shell
+   coder template init --id devcontainer-kubernetes
+   ```
+
+   A list of available templates is shown in the
+   [templates_init](../../../../reference/cli/templates.md) reference.
+
+1. `cd` into the directory and push the template to your Coder deployment:
+
+   ```shell
+   cd $_ && coder templates push
+   ```
+
+   You can also edit the files or make changes to the files before you push them
+   to Coder.
+
+## Registry
+
+1. Go to the
+   [Coder registry](https://registry.coder.com/templates?tag=devcontainer) and
+   select a dev container-compatible template.
+
+1. Copy the files to your local device, then edit them to fit your needs.
+
+1. Upload them to Coder through the CLI or dashboard:
+
+   - CLI:
+
+   ```shell
+   coder templates push <template-name> -d <path to folder containing main.tf>
+   ```
+
+   - Dashboard:
+
+   1. Create a `.zip` of the template files:
+
+      - On Mac or Windows, highlight the files and then right click. A
+        "compress" option is available through the right-click context menu.
+
+      - To zip the files through the command line:
+
+        ```shell
+        zip templates.zip Dockerfile main.tf
+        ```
+
+   1. Select **Templates**.
+   1. Select **Create Template**, then **Upload template**:
+
+      ![Upload template](../../../../images/templates/upload-create-your-first-template.png)
+
+   1. Drag the `.zip` file into the **Upload template** section and fill out the
+      details, then select **Create template**.
+
+      ![Upload the template files](../../../../images/templates/upload-create-template-form.png)
+
+</div>
+
+To set variables such as the namespace, go to the template in your Coder
+dashboard and select **Settings** from the **⋮** (vertical ellipsis) menu:
+
+<Image height="255px" src="../../../../images/templates/template-menu-settings.png" alt="Choose Settings from the template's menu" align="center" />
+
+## Envbuilder Terraform provider
+
+When using the
+[Envbuilder Terraform provider](https://github.com/coder/terraform-provider-envbuilder),
+a previously built and cached image can be reused directly, allowing dev
+containers to start instantaneously.
```

Comment:
Let's use the link in https://registry.terraform.io instead. This has nicer docs

Reply:
https://registry.terraform.io/providers/coder/envbuilder/latest/docs